test/WebJobs.Script.Tests/WebJobsScriptHostServiceTests.cs

```diff
@@ -204,6 +209,50 @@ public async Task HostRestart_DuringInitialization_Recovers()
             Assert.Equal(ScriptHostState.Running, _hostService.State);
         }
 
+        [Fact]
+        public async Task DisposedHost_ServicesNotExposed()
+        {
+            SemaphoreSlim blockingSemaphore = new SemaphoreSlim(0, 1);
+            SemaphoreSlim disposedSemaphore = new SemaphoreSlim(0, 1);
+
+            // Have the first host throw upon starting. Then pause while building the second
+            // host. When accessing Services then, they should be null, rather than disposed.
+            var hostA = CreateMockHost(disposedSemaphore);
+            hostA
+                .Setup(h => h.StartAsync(It.IsAny<CancellationToken>()))
+                .Returns(() =>
+                {
+                    throw new InvalidOperationException("Something happened at startup!");
+                });
+
+            var hostB = CreateMockHost();
+            hostB
+                .Setup(h => h.StartAsync(It.IsAny<CancellationToken>()))
+                .Returns(async () =>
+                {
+                    await blockingSemaphore.WaitAsync();
+                });
+
+            var hostBuilder = new Mock<IScriptHostBuilder>();
+            hostBuilder.SetupSequence(b => b.BuildHost(It.IsAny<bool>(), It.IsAny<bool>()))
+                .Returns(hostA.Object)
+                .Returns(hostB.Object);
+
+            _hostService = new WebJobsScriptHostService(
+               _monitor, hostBuilder.Object, NullLoggerFactory.Instance, _mockRootServiceProvider.Object, _mockRootScopeFactory.Object,
+               _mockScriptWebHostEnvironment.Object, _mockEnvironment.Object, _hostPerformanceManager, _healthMonitorOptions);
+
+            Task startTask = _hostService.StartAsync(CancellationToken.None);
+
+            await disposedSemaphore.WaitAsync();
+
+            Assert.Null(_hostService.Services);
```

Comment:
Before the change, this would not be null. And if you tried to access it, it'd throw an `ObjectDisposedException`.